pymc/aesaraf.py

```diff
@@ -537,23 +537,64 @@ def join_nonshared_inputs(
     point: Dict[str, np.ndarray],
     xs: List[TensorVariable],
     vars: List[TensorVariable],
-    shared,
+    shared: Dict[TensorVariable, TensorSharedVariable],
     make_shared: bool = False,
-):
+) -> Tuple[List[TensorVariable], TensorVariable]:
     """
-    Takes a list of Aesara Variables and joins their non shared inputs into a single input.
+    Takes a list of TensorVariables and joins their non shared inputs into a single input.
 
     Parameters
     ----------
-    point: a sample point
-    xs: list of Aesara tensors
-    vars: list of variables to join
+    point: a sample point associated with a model and variable inputs
+    xs: list of TensorVariable to replace subgraphs with vars and shared
+    vars: list of TensorVariable to reshape, join, and return as inarray
+    shared: dict of TensorVariable and their associated TensorSharedVariable in
+        subgraph replacement
+    make_shared: bool flag to use point argument to build subgraphs of xs_special
```

Comment:
Some suggestion to make it more precise:

```suggestion
    point: dictionary of variable names to numerical values
      Dictionary that maps each input variable name to a numerical value. The values are used to extract the shape of each input variable to establish a correct mapping between the joined and original inputs. The shape of each variable is assumed to be fixed.
    xs: list of TensorVariable
      List of output TensorVariables whose non-shared inputs will be replaced by a joined vector input.
    vars: list of TensorVariable
      List of input TensorVariables which will be replaced by a joined vector input.
    shared: dict of TensorSharedVariable to TensorSharedVariable, optional
      Dictionary that indicates any TensorSharedVariable that should be replaced by new ones.
    make_shared: bool, defaults to False
      Whether to make the joined vector input a shared variable.
```

Reply:
I see you have "optional" for the `shared` argument.  Do you think it would be better to actually have a default then? i.e. None which then initializes an empty dictionary. Then saying `shared` is optional would be a bit more accurate

```python 
def joined_nonshared_inputs(
    ..., 
    shared=None
): 
    # Either at the top
    if shared is None: 
        shared = dict()

    # Or just where it is used
    if shared is not None: 
        replace.update(shared)
```

Reply:
You're right, it's not Optional. Yes we can make it optional like you suggested.